pkg/registry/ociregistry.go

```diff
@@ -0,0 +1,164 @@
+package registry
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+
+	releasev1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
+)
+
+// OCIRegistryClient storage client for an OCI registry.
+type OCIRegistryClient struct {
+	Host        string
+	Project     string
+	CertFile    string
+	Insecure    bool
+	DryRun      bool
+	initialized bool
+	OI          OrasInterface
+	registry    *remote.Registry
+}
+
+var _ StorageClient = (*OCIRegistryClient)(nil)
+
+// NewOCIRegistry create an OCI registry client.
+func NewOCIRegistry(host string, certFile string, insecure bool) *OCIRegistryClient {
+	return &OCIRegistryClient{
+		Host:     host,
+		CertFile: certFile,
+		Insecure: insecure,
+		OI:       &OrasImplementation{},
+	}
+}
+
+// Init registry configuration.
+func (or *OCIRegistryClient) Init() error {
+	if or.initialized {
+		return nil
+	}
+	or.initialized = true
```

Comment:
nit: set this at the end of the method? That way if there is an error, we don't mark it as initialized.

Reply:
Done.